src/js/createStyleguideConfig.js

```diff
@@ -152,28 +162,49 @@ const linkStyleguides = (config, opts) => {
 
 module.exports = (config, options) => {
     // Monitor and limit the depth at which we link styleguides to direct children only.
-    process.env.creatingStyleguideConfig = process.env.creatingStyleguideConfig || 0;
-    process.env.creatingStyleguideConfig += 1;
-
-    const styleguideConfig = linkStyleguides(
-        {
-            webpackConfig: {
-                module: {
-                    rules: [
-                        {
-                            test: /\.jsx?$/,
-                            include: [path.join(process.cwd(), 'src')],
-                            use: {
-                                loader: 'babel-loader',
-                            },
-                        },
-                    ],
+    process.env.creatingStyleguideConfig = `${getCurrentDepth() + 1}`;
+
+    const webpackConfig = {
+        module: {
+            rules: [
+                {
+                    test: /\.jsx?$/,
+                    include: [path.join(process.cwd(), 'src')],
+                    use: {
+                        loader: 'babel-loader',
+                    },
                 },
-            },
-            ...config,
+            ],
         },
-        options
-    );
+    };
+
+    // only the root should alias singletons
+    if (process.env.creatingStyleguideConfig === '1') {
+        webpackConfig.resolve = {
```

Comment:
Let's use the new `getCurrentDepth()` function here?

Reply:
Fixed